### PR TITLE
Removed specific message envelope and unused code

### PR DESCRIPTION
**Pull request by stoyan-zoubev**
[#16] Remove specific message format restriction

- deleted specific message envelope and updated command bus handler.
- reworked command passthrough handler to be generic so that is does not depend on a certain message format.
- renamed AllowedLocalTopicsList and AllowedCloudMessageTypesList settings to PassthroughTelemetryTopic and PassthroughCommandTopic.
- removed unused MessageMapperConfig setting.

Signed-off-by: Stoyan Zoubev <Stoyan.Zoubev@bosch.io>

### CODE REVIEW

**Review by hristobojilov — APPROVED** [review]

File: flags/flags.go [hunk 1]
```diff
@@ -10,16 +10,93 @@
...
+func AddGlobal(f *flag.FlagSet) (configFile *string) {
+	return flags.AddGlobal(f)
+}
+
+// ConfigCheck checks for config file existence.
+func ConfigCheck(logger logger.Logger, configFile string) {
+	flags.ConfigCheck(logger, configFile)
+}
+
+// Add adds the Cloud Agent flags and uses the provided settings to collect the provided values.
```
Comment (e-grigorov) [inline flags/flags.go:41]: - Cloud Agent -> azure connector
- `provided` is duplicated, the wording can be polished a little

File: config/settings.go [hunk 2]
```diff
@@ -22,13 +22,14 @@ import (
...
-	AllowedLocalTopicsList       string `json:"allowedLocalTopicsList"`
-	AllowedCloudMessageTypesList string `json:"allowedCloudMessageTypesList"`
-	IDScope                      string `json:"idScope"`
+	TenantID         string `json:"tenantId"`
+	ConnectionString string `json:"connectionString"`
+	SASTokenValidity string `json:"sasTokenValidity"`
+	IDScope          string `json:"idScope"`
+
+	// TODO: issue #19 - move passthrough topic settings outside AzureSettings
+	PassthroughCommandTopic   string `json:"passthroughCommandTopic"`
```
Comment (e-grigorov) [inline config/settings.go:31]: - the name should be: ...Topics
- the type should be string[] - we already have such examples in suite bootstrapping and software update components

File: flags/flags.go [hunk 3]
```diff
@@ -10,16 +10,93 @@
...
+		"connectionString", def.ConnectionString,
+		"The connection string for connectivity to Azure IoT Hub",
+	)
+	f.StringVar(&settings.SASTokenValidity,
+		flagSASTokenValidity, def.SASTokenValidity,
+		"The validity period for the generated SAS token for device authentication",
+	)
+	f.StringVar(&settings.PassthroughTelemetryTopic,
+		"passthroughTelemetryTopic", def.PassthroughTelemetryTopic,
+		"The list of passthrough telemetry topics the cloud agent listens to.",
```
Comment (e-grigorov) [inline flags/flags.go:59]: - cloud agent -> azure connector
- the description can clarify that these are local topics
- full stop is not used in the end of these flag descriptions

File: flags/flags.go [hunk 4]
```diff
@@ -10,16 +10,93 @@
...
+	)
+	f.StringVar(&settings.PassthroughCommandTopic,
+		"passthroughCommandTopic", def.PassthroughCommandTopic,
+		"The passthrough command topic where all messages from the cloud are forwarded to.",
+	)
+	f.StringVar(&settings.IDScope, flagIDScope, def.IDScope,
+		"ID Scope from Azure Device Provisioning Service",
```
Comment (e-grigorov) [inline flags/flags.go:66]: - ID Scope -> ID scope
- Azure Device Provisioning Service -> Azure Device Provisioning service
- `from` - I guess that it's `for`

File: flags/flags.go [hunk 5]
```diff
@@ -10,16 +10,93 @@
...
+		flagSASTokenValidity, def.SASTokenValidity,
+		"The validity period for the generated SAS token for device authentication",
+	)
+	f.StringVar(&settings.PassthroughTelemetryTopic,
+		"passthroughTelemetryTopic", def.PassthroughTelemetryTopic,
+		"The list of passthrough telemetry topics the cloud agent listens to.",
+	)
+	f.StringVar(&settings.PassthroughCommandTopic,
+		"passthroughCommandTopic", def.PassthroughCommandTopic,
+		"The passthrough command topic where all messages from the cloud are forwarded to.",
```
Comment (e-grigorov) [inline flags/flags.go:63]: - the description can clarify that these are local topics
- full stop is not used in the end of these flag descriptions

File: flags/flags.go [hunk 6]
```diff
@@ -10,16 +10,93 @@
...
+		"ID Scope from Azure Device Provisioning Service",
+	)
+
+	flags.AddLocalBroker(f, &settings.LocalConnectionSettings, &def.LocalConnectionSettings)
+	flags.AddLog(f, &settings.LogSettings, &def.LogSettings)
+
+	flags.AddTLS(f, &settings.TLSSettings, &def.TLSSettings)
+}
+
+// Copy configured all set flag values to map.
```
Comment (e-grigorov) [inline flags/flags.go:75]: configured all set flag values to map -> copies all configured...

File: go.mod [hunk 7]
```diff
@@ -18,9 +16,9 @@ require (
 	github.com/Jeffail/gabs/v2 v2.6.0 // indirect
 	github.com/cenkalti/backoff/v3 v3.0.0 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/eclipse/ditto-clients-golang v0.0.0-20220225085802-cf3b306280d3 // indirect
 	github.com/eclipse/paho.mqtt.golang v1.4.1 // indirect
 	github.com/fsnotify/fsnotify v1.5.1 // indirect
-	github.com/golang/protobuf v1.4.2 // indirect
```
Comment (e-grigorov) [inline go.mod:23]: General note, I guess that the notice file has to be updated.

File: routing/bus/command.go [hunk 8]
```diff
@@ -64,15 +59,10 @@ func CommandBus(router *message.Router,
...
 func (h *commandBusHandler) HandleMessage(msg *message.Message) ([]*message.Message, error) {
-	cloudMessage := &routingmessage.CloudMessage{}
-	if err := json.Unmarshal(msg.Payload, cloudMessage); err != nil {
-		return nil, errors.Wrap(err, "cannot deserialize cloud message")
-	}
-	msg.SetContext(command.SetMessageToContext(msg, cloudMessage))
 	for _, commandHandler := range h.commandHandlers {
-		if util.ContainsString(commandHandler.Topics(), cloudMessage.CommandName) {
-			return commandHandler.HandleMessage(msg)
+		if msg, err := commandHandler.HandleMessage(msg); err == nil {
```
Comment (e-grigorov) [inline routing/bus/command.go:63]: The error is ignored but should not be.

File: routing/message/handlers/command/passthrough_handler.go [hunk 9]
```diff
@@ -13,48 +13,38 @@
...
 	"github.com/eclipse-kanto/azure-connector/config"
-	"github.com/eclipse-kanto/azure-connector/util"
 	"github.com/eclipse-kanto/suite-connector/connector"
-	"github.com/pkg/errors"
 )
 
 const commandPassthroughHandlerName = "command_passthrough_handler"
 
+// A simple command passthrough handler that forwards all cloud-to-device messages from Azure IoT Hub to local MQTT broker on a preconfigured topic.
+// TODO: to be extended in the future as part of issue #21
```
Comment (e-grigorov) [inline routing/message/handlers/command/passthrough_handler.go:25]: Clean up this TODO, if required, it can be put as a comment to the related ticket.

**Review by e-grigorov — CHANGES_REQUESTED** [review]

File: resources/config.json [hunk 10]
```diff
@@ -1,5 +1,4 @@
 {
     "cacert": "/etc/azure-connector/iothub.crt",
```
Comment (e-grigorov) [inline resources/config.json:2]: Could you please fix it to caCert?

**Pull request merged**